src/compiler/parser.ts

```diff
@@ -8936,20 +8965,27 @@ namespace Parser {
                             pushComment(scanner.getTokenText());
                             break;
                     }
-                    previousWhitespace = token() === SyntaxKind.WhitespaceTrivia;
-                    tok = nextTokenJSDoc();
+                    // TODO: nextTokenJSDocBig always returns Identifier, even when that token ends with some whitespace.
+                    // Make this hack less hacky: call a isWhitespace function, and importantly, the state *currently* being SavingComments doesn't mean that the previous call was for a big token
+                    previousWhitespace = token() === SyntaxKind.WhitespaceTrivia || ((state === JSDocState.SavingComments || state === JSDocState.SavingBackticks) && tok === SyntaxKind.Identifier && scanner.getTokenValue().at(-1) === " ");

```

Comment:
Ah, this fails in CI because `.at` is not a thing, and needs to instead be `elementAt`.

Reply:
Use `endsWith`.

Reply:
This code is now in the scanner, where it's more efficient.
Also, fun fact, `getTokenValue` doesn't allocate, `getTokenText` does. But only scanning an Identifier will set getTokenValue; otherwise it gives you the stale text of a previous identifier.

Probably we should deprecate getTokenValue because it's so confusing to use, but (1) there might (still) be a good reason to pre-allocate identifiers (2) I haven't looked where it's used.